docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -173,12 +179,47 @@ With the following validations:
 - `KubeadmControlPlane.Spec.KubeadmConfigSpec` allows mutations required for supporting following use cases:
     - Change of imagesRepository/imageTags (with validation of CoreDNS supported upgrade)
     - Change of node registration options
-    - Change of pre/post kubeadm commands 
+    - Change of pre/post kubeadm commands
     - Change of cloud init files
 
 And the following defaulting:
 
-- `KubeadmControlPlane.Spec.Replicas: 1` 
+- `KubeadmControlPlane.Spec.Replicas: 1`
+
+##### Changes for supporting different upgrade strategies
+
+```go
+type RolloutType string
+
+const (
+  // ScaleDownRolloutType constant is to set rollout strategy type to scale down.
+  ScaleDownRolloutType  RolloutType = "ScaleDown"  
+
+  // ScaleUpRolloutType constant is to set rollout strategy type to scale up.
+  ScaleUpRolloutType  RolloutType = "ScaleUp"
+)
```

Comment:
The PR title says we should support "scale in", while these options offered here say scale down and scale up. To be honest, while I was reading these options I didn't get a sense of what the system is going to do. 

Is there any prior art in terms of naming when rolling out new control planes?